spec/models/chat_message_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ChatMessage do
+  describe '.cook' do
+    it 'does not support HTML tags' do
+      cooked = ChatMessage.cook("<h1>test</h1>")
+
+      expect(cooked).to eq("<p>&lt;h1&gt;test&lt;/h1&gt;</p>")
+    end
+
+    it 'does not support headings' do
+      cooked = ChatMessage.cook("## heading 2")
+
+      expect(cooked).to eq("<p>## heading 2</p>")
+    end
+
+    it 'does not support horizontal rules' do
+      cooked = ChatMessage.cook("---")
+
+      expect(cooked).to eq("<p>---</p>")
+    end
+
+    it 'supports backticks rule' do
+      cooked = ChatMessage.cook("`test`")
+
+      expect(cooked).to eq("<p><code>test</code></p>")
+    end
+
+    it 'supports fence rule' do
+      cooked = ChatMessage.cook(<<~RAW)
+      ```
+      something = test
+      ```
+      RAW
+
+      expect(cooked).to eq(<<~COOKED.chomp)
+      <pre><code>something = test
+      </code></pre>
+      COOKED
+    end
+
+    it 'supports code rule' do
+      cooked = ChatMessage.cook("    something = test")
+
+      expect(cooked).to eq("<pre><code>something = test</code></pre>")
+    end
+
+    it 'supports blockquote rule' do
+      cooked = ChatMessage.cook("> a quote")
+
+      expect(cooked).to eq("<blockquote>\n<p>a quote</p>\n</blockquote>")
+    end
+
+    it 'supports strikethrough rule' do
+      cooked = ChatMessage.cook("~~test~~")
+
+      expect(cooked).to eq("<p><s>test</s></p>")
+    end
+
+    it 'supports emphasis rule' do
+      cooked = ChatMessage.cook("**bold**")
+
+      expect(cooked).to eq("<p><strong>bold</strong></p>")
+    end
+
+    it 'supports link markdown rule' do
+      qa_comment = Fabricate(:qa_comment, raw: "[test link](https://www.example.com)")
```

Comment:
Noticed a test failure for this, this should be a chat message, copy/paste issue I am sure :)